build/build.go

```diff
@@ -595,6 +595,41 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		so.FrontendAttrs["attest:provenance"] = "mode=min,inline-only=true"
 	}
 
+	var prvmode string
+	if v, ok := opt.Attests["attest:provenance"]; ok {
+		if prvmode, err = getProvenanceMode(*v); err != nil {
+			return nil, nil, err
+		}
+	}
+	gitlabels := os.Getenv("BUILDX_GIT_LABELS")
```

Comment:
`BUILDX_GIT_LABELS` should remain as it is you can add a new env `BUILDX_GIT_INFO` that defaults to true.

If `BUILDX_GIT_LABELS` is true then set labels.
If `BUILDX_GIT_INFO` is not false then set `vcs`. Do not check anything about provenance mode.

Reply:
Changed the behavior with your suggestion